processor/transformprocessor/internal/traces/processor_test.go

```diff
@@ -0,0 +1,222 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traces
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+var (
+	TestSpanStartTime      = time.Date(2020, 2, 11, 20, 26, 12, 321, time.UTC)
+	TestSpanStartTimestamp = pdata.NewTimestampFromTime(TestSpanStartTime)
+
+	TestSpanEndTime      = time.Date(2020, 2, 11, 20, 26, 13, 789, time.UTC)
+	TestSpanEndTimestamp = pdata.NewTimestampFromTime(TestSpanEndTime)
+)
+
+func TestProcess(t *testing.T) {
+	tests := []struct {
+		query string
+		want  func(td pdata.Traces)
+	}{
+		{
+			query: `set(attributes["test"], "pass") where name == "operationA"`,
+			want: func(td pdata.Traces) {
+				td.ResourceSpans().At(0).InstrumentationLibrarySpans().At(0).Spans().At(0).Attributes().InsertString("test", "pass")
+			},
+		},
+		{
+			query: `set(attributes["test"], "pass") where resource.attributes["host.name"] == "localhost"`,
+			want: func(td pdata.Traces) {
+				td.ResourceSpans().At(0).InstrumentationLibrarySpans().At(0).Spans().At(0).Attributes().InsertString("test", "pass")
+				td.ResourceSpans().At(0).InstrumentationLibrarySpans().At(0).Spans().At(1).Attributes().InsertString("test", "pass")
+			},
+		},
+		{
+			query: `keep(attributes, "http.method") where name == "operationA"`,
+			want: func(td pdata.Traces) {
+				td.ResourceSpans().At(0).InstrumentationLibrarySpans().At(0).Spans().At(0).Attributes().Clear()
+				td.ResourceSpans().At(0).InstrumentationLibrarySpans().At(0).Spans().At(0).Attributes().InsertString("http.method", "get")
+			},
+		},
+		{
+			query: `set(status.code, 1) where attributes["http.path"] == "/health"`,
+			want: func(td pdata.Traces) {
+				td.ResourceSpans().At(0).InstrumentationLibrarySpans().At(0).Spans().At(0).Status().SetCode(pdata.StatusCodeOk)
+				td.ResourceSpans().At(0).InstrumentationLibrarySpans().At(0).Spans().At(1).Status().SetCode(pdata.StatusCodeOk)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.query, func(t *testing.T) {
+			td := constructTraces()
+			processor, err := NewProcessor([]string{tt.query}, component.ProcessorCreateSettings{})
+			assert.NoError(t, err)
+
+			_, err = processor.ProcessTraces(context.Background(), td)
+			assert.NoError(t, err)
+
+			exTd := constructTraces()
+			tt.want(exTd)
+
+			assert.Equal(t, exTd, td)
+		})
+	}
+}
+
+func BenchmarkTwoSpans(b *testing.B) {
```

Comment:
Please add the benchmark results to the description. May be good to also compare this with current "span" and "attribute" processor.